helm/v1beta1/types_helm.go

```diff
@@ -0,0 +1,87 @@
+package v1beta1
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:plural=helmchartrepositories
+
+// HelmChartRepository holds cluster-wide configuration for proxied Helm chart repository
+type HelmChartRepository struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec HelmChartRepositorySpec `json:"spec"`
+
+	// Observed status of the repository within the cluster..
+	// +optional
+	Status HelmChartRepositoryStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type HelmChartRepositoryList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []HelmChartRepository `json:"items"`
+}
+
+// Helm chart repository exposed within the cluster
+type HelmChartRepositorySpec struct {
+
+	// Optional associated human readable repository name, it can be used by UI for displaying purposes
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	DisplayName string `json:"name,omitempty"`
+
+	// Optional human readable repository description, it can be used by UI for displaying purposes
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// Required configuration for connecting to the chart repo
+	ConnectionConfig `json:",inline"`
```

Comment:
This a related, separate stanza. It should be nested like https://github.com/kubernetes/api/blob/master/admissionregistration/v1/types.go#L184  `connectionConfig` is appropriate

Reply:
what speaks against the embedded struct? URL is something that needs to be specified always by user, other things much less. With that we will always have a additional nesting level. Is that user-friendly?

Reply:
> With that we will always have a additional nesting level. Is that user-friendly?

Yes. As the resource grows (and even in its current form), the connection information ends up tightly grouped when you're looking at no matter how it is sorted. It's more user friendly for human inspection and review.